consensus/src/threshold_simplex/actors/voter/actor.rs

```diff
@@ -1934,6 +2058,163 @@ impl<
         // Create shutdown tracker
         let mut shutdown = self.context.stopped();
 
+        // Spawn partial signature verifier
+        //
+        // We attempt to verify notarization/nullification/finalization messages right away (outside this loop)
+        // in case they help us avoid unnecessary work.
+        let (mut verifier_sender, mut verifier_receiver) =
+            mpsc::channel::<(View, View, C::PublicKey, Voter<V, D>)>(1024);
+        let (mut verified_sender, mut verified_receiver) =
+            mpsc::channel::<(C::PublicKey, Voter<V, D>)>(1024);
+        self.context.with_label("verifier").spawn({
+            let mut latest = observed_view;
+            let namespace = self.namespace.clone();
+            let supervisor = self.supervisor.clone();
+            let batch_size = self.batch_size.take().unwrap();
+            move |_| async move {
+                // Initialize view data structures
+                let mut counter: usize = 0;
+                let mut min_view: View = 0;
+                let mut messages = HashMap::new();
+                #[allow(clippy::type_complexity)]
+                let mut work: BTreeMap<
+                    u64,
+                    BTreeMap<(Vec<u8>, Vec<u8>), HashMap<PartialSignature<V>, Vec<usize>>>,
+                > = BTreeMap::new();
+
+                loop {
+                    // Read at least one message
+                    if !recv_batch(
+                        &mut verifier_receiver,
+                        work.is_empty(),
+                        |(min, current, sender, msg)| {
+                            min_view = min;
+                            latest = current;
+                            let view = msg.view();
+                            let verifiable = msg.message(&namespace);
+                            messages.insert(counter, (sender, msg, verifiable.len()));
+                            for (namespace, message, signature) in verifiable {
+                                work.entry(view)
+                                    .or_default()
+                                    .entry((namespace, message))
+                                    .or_default()
+                                    .entry(signature)
+                                    .or_default()
+                                    .push(counter);
+                            }
+                            counter = counter.wrapping_add(1);
+                        },
+                    )
+                    .await
+                    {
+                        return;
+                    }
+
+                    // Attempt to do something in current view
+                    let (view, (namespace, message), signatures) =
```

Comment:
Verify message with seed (otherwise can have delay for a message to be processed)